backend/src/models/membership.go

```diff
@@ -0,0 +1,29 @@
+package models
+
+import "github.com/google/uuid"
+
+type MembershipType string
+
+const (
+	MembershipTypeMember MembershipType = "member"
+	MembershipTypeAdmin MembershipType = "admin"
+)
+
+type Tabler interface {
+	TableName() string
+}
+
+func (Membership) TableName() string {
+	return "user_club_members"
+}
+
+type Membership struct {
```

Comment:
for logical organization, should the id be on the line above the pointer to the corresponding entity?

Reply:
like this 
```suggestion
type Membership struct {
	Model

	ClubID uuid.UUID `gorm:"primaryKey" json:"club_id" validate:"required,min=1"`
	Club *Club `gorm:"constraint:OnUpdate:CASCADE,OnDelete:CASCADE;" json:"-" validate:"-"`

	UserID uuid.UUID `gorm:"primaryKey" json:"user_id" validate:"required,min=1"`
	User *User `gorm:"constraint:OnUpdate:CASCADE,OnDelete:CASCADE;" json:"-" validate:"-"`

	Type MembershipType `gorm:"type:varchar(255);default:member" json:"membership_type" validate:"required,max=255"`
}
```